File: app/models/topic_tree_path.rb
# frozen_string_literal: true
class TopicTreePath < ApplicationRecord
  validates :ancestor_id,
            presence: true,
            uniqueness: { scope: :descendant_id }

  validates :descendant_id, presence: true
  validates :parental_flg, inclusion: { in: [true, false] }
end

File: frontend/react/stores/entities/TopicTreePath.js
import { Record } from 'immutable';

const TopicTreePathBase = new Record({
  ancestorId: null,
  descendantId: null,
  parentalFlg: false
});

export default class TopicTreePath extends TopicTreePathBase {
  /**
   * 直接の親子関係かどうかを表す
   */
  isParental() {
    return this.parentalFlg;
  }
}
